s64da_benchmark_toolkit/prepare.py

```diff
@@ -111,11 +111,19 @@ def _run_tasks_parallel(self, tasks):
         # randomize the tasks to decrease lock contention
         random.shuffle(tasks)
         with ThreadPoolExecutor(max_workers=self.args.max_jobs) as executor:
-            futures = [executor.submit(self._run_shell_task, task) for task in tasks]
+            futures = []
+            for task in tasks:
+                if callable(task):
+                    future = executor.submit(task)
+                else:
+                    future = executor.submit(self._run_shell_task, task)
+                futures.append(future)
+
```

Comment:
How about this implementation:
```
    def _run_tasks_parallel(self, tasks):
        def get_future(task):
            if callable(task):
                return executor.submit(task)
            else:
                return executor.submit(self._run_shell_task, task)

        # randomize the tasks to decrease lock contention
        random.shuffle(tasks)
        with ThreadPoolExecutor(max_workers=self.args.max_jobs) as executor:
            futures = [get_future(task) for task in tasks]
```

Just a suggestion

Reply:
Love the idea, changed accordingly. So much pythonic.